libraries/AP_SmartAudio/AP_SmartAudio.cpp

```diff
@@ -0,0 +1,955 @@
+    }
+}
+
+
+
+// updates the smartaudio state in sync whith AP_VideoTX
+bool AP_SmartAudio::update(bool force)
+{
+
+
+    if(vtx_states_queue.is_empty() && requests_queue.is_empty() && !is_waiting_response){
+        if(!hal.util->get_soft_armed()) {
+            request_settings();
+        }
+        return false;
+    }
+    smartaudioSettings_t _current_state;
+    _get_current_state(&_current_state);
+     
+     debug("is_waiting_response:%u",is_waiting_response);
+    if((is_waiting_response || _get_current_state(&_current_state)==nullptr) && !force){
+        debug("Breaking the update ...");
+        return false;
+    }
+    _print_state(&_current_state);
+
+    // send request update for freq with ap_vtx values
+    if(_current_state.userFrequencyMode && _current_state.frequency!=ap_video_tx->get_frequency_mhz()){
+        debug("UPDATE AP_VTX->HW_VTX: FREQ");
+        set_frequency(ap_video_tx->get_frequency_mhz(),_current_state.pitModeRunning);
+    }
+
+    // send request update channel  with ap_vtx values
+    if(!_current_state.userFrequencyMode && (_current_state.channel!=ap_video_tx->get_channel() || _current_state.band!=ap_video_tx->get_band())){
+        debug("UPDATE AP_VTX->HW_VTX: CHAN");
+        set_channel((ap_video_tx->get_band()*8)+ap_video_tx->get_channel());
+    }
+
+    // send request update for power with ap_vtx values
+    if(_current_state.version==SMARTAUDIO_SPEC_PROTOCOL_v21 && _get_power_in_mw_from_dbm(_current_state.power_in_dbm)!= ap_video_tx->get_power_mw()){
+        debug("UPDATE AP_VTX->HW_VTX: POW IN MW FROM DBM");
+        set_power_mw(ap_video_tx->get_power_mw());
+    }
+
+    // send request update for power with ap_vtx values
+    if(_current_state.version!=SMARTAUDIO_SPEC_PROTOCOL_v21
+    && ap_video_tx->get_power_mw()!=_get_power_in_mw_from_dbm(_get_power_in_dbm_from_vtx_power_level(_current_state.power,_current_state.version))){
+        debug("UPDATE AP_VTX->HW_VTX: POW IN MW FROM PWLEVEL %d -> %d"
+        ,ap_video_tx->get_power_mw()
+        ,_get_power_in_mw_from_dbm(_get_power_in_dbm_from_vtx_power_level(_current_state.power,_current_state.version))
+        );
+
+        set_power_mw(ap_video_tx->get_power_mw());
+    }
+
+    // send request update for options with ap_vtx values
+    if( (ap_video_tx->get_options()==0 && _current_state.pitModeRunning) || (ap_video_tx->get_options()==1 && !_current_state.pitModeRunning) ){
+        debug("UPDATE AP_VTX->HW_VTX: OPTIONS");
+        uint8_t operation_mode= 0x00;
+
+        debug("UPDATE AP_VTX->HW_VTX: OPTIONS LOCKING %u %u",ap_video_tx->get_locking(),ap_video_tx->get_locking()<<3);
+        debug("OPERATION MODE TO UPDATE:%u %u %u %u"
+        ,0x0F & ap_video_tx->get_locking()<<3
+        ,0X0F & ap_video_tx->get_options()<<2
+        ,0X0F & _current_state.pitmodeOutRangeActive<<1
+        ,0X0F & _current_state.pitmodeInRangeActive<<0
+        );
+
+        operation_mode |= _current_state.pitmodeInRangeActive<<0;
```

Comment:
Brackets around the bitshift operation

Reply:
Fixed in 8d208cdac75896b4404257600404fb1403fa4e8f